tests/test_hmc.py

```diff
@@ -0,0 +1,62 @@
+import aesara
+import aesara.tensor as aet
+import numpy as np
+from aesara.tensor.random.utils import RandomStream
+from aesara.tensor.var import TensorVariable
+
+import aesara_hmc.hmc as hmc
+
+
+def test_hmc():
+    def potential_fn(q: TensorVariable) -> TensorVariable:
+        return -1.0 / aet.power(aet.square(q[0]) + aet.square(q[1]), 0.5)
+
+    srng = RandomStream(seed=59)
+
+    step_size = aet.scalar("step_size")
+    inverse_mass_matrix = aet.vector("inverse_mass_matrix")
+    num_integration_steps = aet.scalar("num_integration_steps", dtype="int32")
+
+    q = aet.vector("q")
+    potential_energy = potential_fn(q)
+    potential_energy_grad = aesara.grad(potential_energy, wrt=q)
+
+    kernel = hmc.kernel(
+        srng, potential_fn, step_size, inverse_mass_matrix, num_integration_steps
+    )
+    next_step = kernel(q, potential_energy, potential_energy_grad)
+
+    # Compile a function that returns the next state
+    step_fn = aesara.function(
+        (q, step_size, inverse_mass_matrix, num_integration_steps), next_step
+    )
+
+    # Compile a function that integrates the trajectory integrating several times
+    trajectory, _ = aesara.scan(
+        fn=kernel,
+        outputs_info=[
+            {"initial": q},
+            {"initial": potential_energy},
+            {"initial": potential_energy_grad},
+        ],
+        n_steps=1000,
+    )
+    traj = aesara.function(
+        (q, step_size, inverse_mass_matrix, num_integration_steps), trajectory
+    )
```

Comment:
@brandonwillard while integrating with the `for` loop below works fine, I cannot make this one work. 

This happens during compilation:

<details>
<summary>Stack trace</summary>

```
../../../.virtualenvs/aesara-hmc/lib/python3.9/site-packages/aesara/compile/function/__init__.py:337: in function
    fn = pfunc(
../../../.virtualenvs/aesara-hmc/lib/python3.9/site-packages/aesara/compile/function/pfunc.py:524: in pfunc
    return orig_function(
../../../.virtualenvs/aesara-hmc/lib/python3.9/site-packages/aesara/compile/function/types.py:1972: in orig_function
    m = Maker(
../../../.virtualenvs/aesara-hmc/lib/python3.9/site-packages/aesara/compile/function/types.py:1586: in __init__
    fgraph, additional_outputs = std_fgraph(inputs, outputs, accept_inplace)
../../../.virtualenvs/aesara-hmc/lib/python3.9/site-packages/aesara/compile/function/types.py:190: in std_fgraph
    fgraph = FunctionGraph(orig_inputs, orig_outputs, update_mapping=update_mapping)
../../../.virtualenvs/aesara-hmc/lib/python3.9/site-packages/aesara/graph/fg.py:167: in __init__
    self.import_var(output, reason="init")
../../../.virtualenvs/aesara-hmc/lib/python3.9/site-packages/aesara/graph/fg.py:336: in import_var
    self.import_node(var.owner, reason=reason, import_missing=import_missing)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = FunctionGraph(Subtensor{int64::}(*29 -> for{cpu,scan_fn}(TensorConstant{1000}, *23 -> IncSubtensor{Set;:int64:}(AllocE...nt{1}), ScalarConstant{-1}))), *14))))), TensorConstant{-inf}, *35)), TensorConstant{0}, TensorConstant{1.0})), *28::0)
apply_node = bernoulli_rv(RandomStateSharedVariable(<RandomState(MT19937) at 0x7F712F0DE540>), TensorConstant{[]}, TensorConstant{4}, Elemwise{clip,no_inplace}.0)
check = True, reason = 'init', import_missing = False

    def import_node(
        self,
        apply_node: Apply,
        check: bool = True,
        reason: str = None,
        import_missing: bool = False,
    ) -> None:
        """Recursively import everything between an `Apply` node and the `FunctionGraph`'s outputs.
    
        Parameters:
        ----------
        apply_node : aesara.graph.basic.Apply
            The node to be imported.
        check : bool
            Check that the inputs for the imported nodes are also present in
            the `FunctionGraph`.
        reason : str
            The name of the optimization or operation in progress.
        import_missing : bool
            Add missing inputs instead of raising an exception.
        """
        # We import the nodes in topological order. We only are interested in
        # new nodes, so we use all variables we know of as if they were the
        # input set.  (The functions in the graph module only use the input set
        # to know where to stop going down.)
        new_nodes = io_toposort(self.variables, apply_node.outputs)
    
        if check:
            for node in new_nodes:
                for var in node.inputs:
                    if (
                        var.owner is None
                        and not isinstance(var, Constant)
                        and var not in self.inputs
                    ):
                        if import_missing:
                            self.add_input(var)
                        else:
                            error_msg = (
                                f"Input {node.inputs.index(var)} ({var})"
                                " of the graph (indices start "
                                f"from 0), used to compute {node}, was not "
                                "provided and not given a value. Use the "
                                "Aesara flag exception_verbosity='high', "
                                "for more information on this error."
                            )
>                           raise MissingInputError(error_msg, variable=var)
E                           aesara.graph.fg.MissingInputError: Input 0 (<TensorType(float64, vector)>) of the graph (indices start from 0), used to compute InplaceDimShuffle{x,0}(<TensorType(float64, vector)>), was not provided and not given a value. Use the Aesara flag exception_verbosity='high', for more information on this error.
E                            
E                           Backtrace when that variable is created:
E                           
E                             File "/home/remi/.virtualenvs/aesara-hmc/lib/python3.9/site-packages/_pytest/runner.py", line 162, in pytest_runtest_call
E                               item.runtest()
E                             File "/home/remi/.virtualenvs/aesara-hmc/lib/python3.9/site-packages/_pytest/python.py", line 1641, in runtest
E                               self.ihook.pytest_pyfunc_call(pyfuncitem=self)
E                             File "/home/remi/.virtualenvs/aesara-hmc/lib/python3.9/site-packages/pluggy/hooks.py", line 286, in __call__
E                               return self._hookexec(self, self.get_hookimpls(), kwargs)
E                             File "/home/remi/.virtualenvs/aesara-hmc/lib/python3.9/site-packages/pluggy/manager.py", line 93, in _hookexec
E                               return self._inner_hookexec(hook, methods, kwargs)
E                             File "/home/remi/.virtualenvs/aesara-hmc/lib/python3.9/site-packages/pluggy/manager.py", line 84, in <lambda>
E                               self._inner_hookexec = lambda hook, methods, kwargs: hook.multicall(
E                             File "/home/remi/.virtualenvs/aesara-hmc/lib/python3.9/site-packages/pluggy/callers.py", line 187, in _multicall
E                               res = hook_impl.function(*args)
E                             File "/home/remi/.virtualenvs/aesara-hmc/lib/python3.9/site-packages/_pytest/python.py", line 183, in pytest_pyfunc_call
E                               result = testfunction(**testargs)
E                             File "/home/remi/projects/ampersand/aehmc/tests/test_hmc.py", line 35, in test_hmc
E                               trajectory, _ = aesara.scan(

../../../.virtualenvs/aesara-hmc/lib/python3.9/site-packages/aesara/graph/fg.py:401: MissingInputError
```

</details>

Any idea?